endpoints/openrtb2/auction.go

```diff
@@ -1282,14 +1287,45 @@ func effectivePubID(pub *openrtb.Publisher) string {
 	return pbsmetrics.PublisherUnknown
 }
 
-func validateAccount(cfg *config.Configuration, pubID string) error {
-	var err error = nil
-	if cfg.AccountRequired && pubID == pbsmetrics.PublisherUnknown {
-		// If specified in the configuration, discard requests that don't come with an account ID.
-		err = error(&errortypes.AcctRequired{Message: fmt.Sprintf("Prebid-server has been configured to discard requests that don't come with an Account ID. Please reach out to the prebid server host.")})
-	} else if _, found := cfg.BlacklistedAcctMap[pubID]; found {
-		// Blacklist account now that we have resolved the value
-		err = error(&errortypes.BlacklistedAcct{Message: fmt.Sprintf("Prebid-server has blacklisted Account ID: %s, please reach out to the prebid server host.", pubID)})
+func (deps *endpointDeps) getAccount(ctx context.Context, pubID string) (*config.Account, error) {
+	// Check BlacklistedAcctMap until we have deprecated it
+	if _, found := deps.cfg.BlacklistedAcctMap[pubID]; found {
+		return nil, error(&errortypes.BlacklistedAcct{
+			Message: fmt.Sprintf("Prebid-server has disabled Account ID: %s, please reach out to the prebid server host.", pubID),
+		})
 	}
-	return err
+	var account *config.Account
+	var accountJSON json.RawMessage
+	var err error
+	if accountJSON, err = deps.accounts.FetchAccount(ctx, pubID); err != nil || accountJSON == nil {
+		// pubID does not reference a valid account, or is empty
+		if deps.cfg.AccountRequired &&
```

Comment:
Will change.

The assumption I made was that requests without a publisher ID should be rare. As it so happens, this made the code more compact as that error is only generated in one place (with the precheck, it would be in two places).